test/Test/Data/Map.purs

```diff
@@ -339,3 +339,41 @@ mapTests = do
   quickCheck \(TestMap m :: TestMap Int Int) ->
     let outList = foldrWithIndex (\i a b -> (Tuple i a) : b) Nil m
     in outList == sort outList
+
+  -- Identity: (pure identity) <*> v = v
+  log "applyWithDefault abides applicative laws: Identity"
+  quickCheck \(TestMap x :: TestMap Int Int) ->
+    let out = M.applyWithDefault M.empty (Just identity) x Nothing
+    in out == x
+
+  -- Composition: pure (<<<) <*> f <*> g <*> h = f <*> (g <*> h)
+  log "applyWithDefault abides applicative laws: Composition"
+  quickCheck \(TestMap f :: TestMap Int (Boolean -> String -> Int))
+              fd
+              (TestMap g :: TestMap Int Boolean)
+              gd
+              (TestMap h :: TestMap Int String)
+              hd ->
+    let left0 = M.applyWithDefault M.empty (Just identity) f fd
+        left1 = M.applyWithDefault left0 fd g gd
+        left2 = M.applyWithDefault left1 (fd <*> gd) h hd
+        right0 = M.applyWithDefault f fd g gd
+        right1 = M.applyWithDefault right0 (fd <*> gd) h hd
+    in left2 == right1
```

Comment:
Is this checking associativity? I haven't tried to look if it's an equivalent formulation, but it looks like the values are grouped the same way. Is this the associative grouping?

```PureScript
let left0 = M.applyWithDefault M.empty (Just compose) f fd
    left1 = M.applyWithDefault left0 (Just compose <*> fd) g gd
    left2 = M.applyWithDefault left1 (Just compose <*> fd <*> gd) h hd
    right0 = M.applyWithDefault g gd h hd
    right1 = M.applyWithDefault f fd right0 (gd <*> hd)
in left2 == right1
```

I haven't checked that either.

Reply:
You're right, I completely messed this up 😅